Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -260,20 +262,20 @@ - (void)listenToTargetWithQueryData:(FSTQueryData *)queryData {
 }
 
 - (void)sendWatchRequestWithQueryData:(FSTQueryData *)queryData {
-  [self.watchChangeAggregator recordTargetRequest:@(queryData.targetID)];
+  [self.watchChangeAggregator recordTargetRequest:queryData.targetID];
   _watchStream->WatchQuery(queryData);
 }
 
 - (void)stopListeningToTargetID:(TargetId)targetID {
-  FSTBoxedTargetID *targetKey = @(targetID);
-  FSTQueryData *queryData = self.listenTargets[targetKey];
-  HARD_ASSERT(queryData, "stopListeningToTargetID: target not currently watched: %s", targetKey);
+  auto found = _listenTargets.find(targetID);
```

Comment:
Looks good. Alternative:

```
size_t num_removed = _listenTargets.erase(targetId);
HARD_ASSERT(num_removed == 1, ...);
```

(You could even inline num_removed... but inlining side effects into asserts makes me nervous, even though this assert is never compiled out.)